### ISSUE TITLE
the Random Forest doesn't work

### ISSUE THREAD

**Issue by AlanSpencer2** [Xtra-Computing/thundergbm#68]
I am new to thundergbm, and just trying to get a simple Random Forest classifier going. But the classifier classifies every sample to be 1, not one single 0 out of 188244 samples. No other classifier behaves like this. I also tried different number of trees, depth etc. But it still classies everything to 1. Is there something wrong with the following code?

from thundergbm import TGBMClassifier
clf = TGBMClassifier(depth=6, n_trees = 1, n_parallel_trees=100, bagging=1)
clf.fit(X_train, y_train)
y_pred = clf.predict(X_test)

#y_pred classifies everything in the test set (X_test) to one.

**Comment by Kurt-Liuhf** [comment]
@AlanSpencer2 Hi, I used the classifier with the same parameters to fit the `covtype` data set from `sklearn` but I could not reproduce your results. The predictions seem to be correct. So it would be better if you can provide a subset of your data set.
Thanks.

**Comment by AlanSpencer2** [comment]
Hi, the problem occurs when the label/target variable is binary 0 or 1, True or False. Not when there are multiple classes. Can you please try a binary classification problem?

Here is the iris dataset with 3 different flower types. The label is 1 if the flower is of type setosa, and 0 for the other flower types:
[iris_data.csv](https://github.com/Xtra-Computing/thundergbm/files/7790305/iris_data.csv)

The Python code:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from thundergbm import TGBMClassifier

df = pd.read_csv(r'C:\Python\iris_data.csv', encoding='ISO-8859-1', low_memory=False, index_col=0)
df
X=df[['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']] 
y=df['Label'] 
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3)
clf = TGBMClassifier(depth=6, n_trees = 1, n_parallel_trees=100, bagging=1)
clf.fit(X_train, y_train)
pred_test = clf.predict(X_test)
